Reference/Common/Library/repository_rest.php

```diff
@@ -210,4 +210,40 @@ function _repository_list_files ($uri) {
 	return $response;
 }
 
-?>
\ No newline at end of file
+/**
+ * curl -u ncroot:NCROOT_PWD  -XGET "http://localhost:10080/ubi-api-rest/repository/v2/resource/workflow?uri={uri}"
+ */
+function _repository_get_workflow_definition ($uri) {
+        $msa_rest_api = "repository/v2/resource/workflow?uri={$uri}";
+        $curl_cmd = create_msa_operation_request(OP_GET, $msa_rest_api);
+        $response = perform_curl_operation($curl_cmd, "REPOSITORY GET WORKFLOW DEFINITION");
+        $response = json_decode($response, true);
+        if ($response['wo_status'] !== ENDED) {
+                $response = json_encode($response);
+                return $response;
+        }
+        $response = prepare_json_response(ENDED, ENDED_SUCCESSFULLY, $response['wo_newparams']['response_body']);
+        return $response;
+}
+
+/**
+ * curl -u ncroot:NCROOT_PWD  -XPOST "http://localhost:10080/ubi-api-rest/repository/v2/resource/workflow?uri={uri} -d '{"content": "xyz"}'"
+ */
+function _repository_change_workflow_definition ($uri, $content) {
+        $msa_rest_api = "repository/v2/resource/workflow?uri={$uri}";
+        $content_array = json_decode($content, True);
+        if (!$content_array['example']) {
```

Comment:
This line can generate a PHP Notice.
It should be better to test 
if (!isset($content_array['example'])) {

By the way why 'example' has to be present in $content_array() ?

Reply:
could be related to the "example" section of a WF definition file:
![image](https://user-images.githubusercontent.com/7395473/118959850-fbf4f380-b962-11eb-94c7-1137c5a9f08f.png)